server/token.ts

```diff
@@ -0,0 +1,42 @@
+import { env, exit } from "process";
+import api from "../common/api";
+
+const savedTokens: string[] = [];
+let index = 0;
+
+export const initTokenFromEnv = async () => {
+  const envTokenString = env.TOKEN;
+  if (!envTokenString) {
+    console.error("Token not found");
+    exit(-1);
+  }
+
+  const tokens = envTokenString.split(",");
+  // Check token usable
+  for (const token of tokens) {
+    try {
+      await api.getRepoStargazersCount("bytebase/star-history", token);
+      savedTokens.push(token);
+    } catch (error) {
+      console.error(`token is unusable: ${token}`);
+    }
+  }
+
+  if (savedTokens.length === 0) {
+    console.error("No token");
+    exit(-1);
+  }
+
+  console.log(`Usable token amount: ${savedTokens.length}`);
+};
+
+export const getNextToken = () => {
+  const currentIndex = index;
+
+  index++;
```

Comment:
Looks like we would run into a concurrency issue.

Thread 1 at Line 35
Thread 2 at Line 40
Thread 1 executes index++ at line 36
Thread 2 at 41 will have out of bound index

better use currentIndex % savedTokens.length as the index